### PR TITLE
refactor(header): consolidate header actions into a dropdown menu and…

### LINKED ISSUE #109: feat : add clear history functionality
You should be able to clear history

I have opened a pr, please do review

@piyushgarg-dev

### PR DESCRIPTION
**Pull request by ruru-m07**
# TL;DR

This pull request introduces several updates and new features to the project, particularly focusing on the integration of the `@radix-ui/react-dropdown-menu` and enhancing the dropdown menu functionality in the user interface. The most important changes include updating dependencies, adding a new dropdown menu component, and modifying the chat interface to incorporate the new dropdown menu. Additionally, there are updates to the IndexedDB utility functions and schema definitions.

### Dependency Updates:
* Added `@radix-ui/react-dropdown-menu` to `package.json` and `pnpm-lock.yaml` to enable dropdown menu functionality. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R18) [[2]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR20-R22) [[3]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1048-R1073) [[4]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1151-R1187) [[5]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1279-R1306)

### New Components:
* Created a new dropdown menu component in `src/components/ui/dropdown-menu.tsx` using `@radix-ui/react-dropdown-menu`.

### UI Enhancements:
* Replaced the select component with the new dropdown menu in the chat interface, and added functionality to clear chat history using the dropdown menu. [[1]](diffhunk://#diff-2543abb4a029897a3f3506253727bde95036fa5c023acf5c0b270ed9c8bed2c3L3-R10) [[2]](diffhunk://#diff-2543abb4a029897a3f3506253727bde95036fa5c023acf5c0b270ed9c8bed2c3R43-R58) [[3]](diffhunk://#diff-2543abb4a029897a3f3506253727bde95036fa5c023acf5c0b270ed9c8bed2c3R114-R121) [[4]](diffhunk://#diff-2543abb4a029897a3f3506253727bde95036fa5c023acf5c0b270ed9c8bed2c3L252-R334)

### IndexedDB Utility Updates:
* Added a new `clearChatHistory` function to `src/lib/indexedDB.ts` and updated `useIndexDB` hook to include this function. [[1]](diffhunk://#diff-9a8623dbf7e770f57559b4a5b4570a98d97042cb08b165ff120aff574c258156R49-R53) [[2]](diffhunk://#diff-3e8bfb70a7808720ec21426d70188254b536fe3d8c87ae447f03d399a938fb00L2-R6) [[3]](diffhunk://#diff-3e8bfb70a7808720ec21426d70188254b536fe3d8c87ae447f03d399a938fb00R21-R24)

### Schema Updates:
* Extended the supported programming languages in the `outputSchema` in `src/schema/modeOutput.ts`. [[1]](diffhunk://#diff-51a7f1b4dce9a44ca6bd3869a05d2bd17d7b588d314a64f03ebad37c2ab7cb01R3-R29) [[2]](diffhunk://#diff-51a7f1b4dce9a44ca6bd3869a05d2bd17d7b588d314a64f03ebad37c2ab7cb01L12-R39)… add clear functionality

close #109 

image

### CODE REVIEW

**Comment by ayush-oswal** [comment]
@ruru-m07 bhai muje bhi toh mauka do contribution ka 😅😅

**Comment by ruru-m07** [comment]
PepeCRY~1

@ayush-oswal

**Comment by ruru-m07** [comment]
@piyushgarg-dev yes boss. good to go
